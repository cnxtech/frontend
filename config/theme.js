import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
