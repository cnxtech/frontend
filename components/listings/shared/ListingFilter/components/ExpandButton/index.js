import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import theme from 'config/theme'
import {
  StyledExpandButton,
  ArrowIcon
} from './styles'

class ExpandButton extends PureComponent {
  render() {
    return (
      <StyledExpandButton onClick={this.props.onClick}>
        <ArrowIcon name={this.props.expanded ? "chevron-up" : "chevron-down"} color={theme.colors.dark} size={16} />
      </StyledExpandButton>
    )
  }
}

ExpandButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  expanded: PropTypes.bool.isRequired
}

export default ExpandButton
