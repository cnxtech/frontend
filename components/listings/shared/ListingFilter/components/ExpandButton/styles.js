import styled from 'styled-components'
import theme from 'config/theme'
import Button from '@emcasa/ui-dom/components/Button'
import Icon from '@emcasa/ui-dom/components/Icon'

const EXPAND_BUTTON_SIZE = 32

const StyledExpandButton = styled(Button)`
  position: fixed;
  right: 0;
  border: 1px solid ${theme.colors.lightGrey};
  color: ${({active}) => active ? theme.colors.white : theme.colors.pink};
  font-size: ${theme.fontSizes[1]}px;
  margin: 0 ${theme.space[4]}px ${theme.space[1]}px 0;
  width: ${EXPAND_BUTTON_SIZE}px;
  height: ${EXPAND_BUTTON_SIZE}px;

  display: flex;
  justify-content: center;
  align-items: center;

  @media screen and (min-width: 768px) {
    display: none;
  }
`

const ArrowIcon = styled(Icon)`
  display: flex;
  align-items: center;
  height: ${EXPAND_BUTTON_SIZE / 2}px;
`

export {
  StyledExpandButton,
  ArrowIcon,
  EXPAND_BUTTON_SIZE
}
