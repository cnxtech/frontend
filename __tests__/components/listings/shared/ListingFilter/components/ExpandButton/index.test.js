import React from 'react'
import ExpandButton from 'components/listings/shared/ListingFilter/components/ExpandButton'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

describe('<ExpandButton/>', () => {
  it('should render component', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <ExpandButton
          expanded={false}
          onClick={() => {}}
        />
      </ThemeProvider>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
