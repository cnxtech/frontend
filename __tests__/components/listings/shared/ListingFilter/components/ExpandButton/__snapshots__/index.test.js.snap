// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandButton/> should render component 1`] = `
.c1 {
  outline: none;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-color: #e0e6ed;
  background-color: #ffffff;
  border-radius: 4px;
  border-width: 1px;
  width: auto;
  height: 40px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-style: solid;
}

.c1 :hover {
  background-color: #ffffff;
}

.c0 {
  position: fixed;
  right: 0;
  border: 1px solid #e0e6ed;
  color: #f50057;
  font-size: 14px;
  margin: 0 20px 4px 0;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 16px;
}

@media screen and (min-width:768px) {
  .c0 {
    display: none;
  }
}

<button
  className="c0 c1"
  fontSize="default"
  height="medium"
  onClick={[Function]}
>
  <div
    className="c2 "
  >
    <svg
      height={20}
      version="1.1"
      viewBox="0 0 320 512"
      x={0}
      y={0}
    >
      <path
        d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
        name="angle-down"
        type="default"
      />
    </svg>
  </div>
</button>
`;
